src/charm.py

```diff
@@ -245,6 +247,10 @@ def _on_pfcp_agent_pebble_ready(self, event: EventBase) -> None:
         """Handle pfcp agent Pebble ready event."""
         if not self.unit.is_leader():
             return
+        if not self._pfcp_agent_container.can_connect():
+            self.unit.status = WaitingStatus("Waiting for pfcp agent container to be ready")
+            event.defer()
```

Comment:
Here we can remove the `defer()`, this handler will be called again by `PebbleReady`.